platform-modules/mimetype-manager/src/main/scala/org/sunbird/mimetype/mgr/BaseMimeTypeManager.scala

```diff
@@ -65,10 +65,10 @@ class BaseMimeTypeManager(implicit ss: StorageService) {
 			throw new ClientException("ERR_INVALID_FILE_URL", "Please Provide Valid File Url!")
 	}
 
-	def uploadArtifactToCloud(uploadedFile: File, identifier: String): Array[String] = {
+	def uploadArtifactToCloud(uploadedFile: File, identifier: String, filePath: Option[String] = None): Array[String] = {
 		var urlArray = new Array[String](2)
 		try {
-			val folder = Platform.getString(CONTENT_FOLDER, "content") + File.separator + Slug.makeSlug(identifier, true) + File.separator + Platform.getString(ARTIFACT_FOLDER, "artifact")
+			val folder = if(StringUtils.isNotBlank(filePath.getOrElse(""))) filePath.get + File.separator + Platform.getString(CONTENT_FOLDER, "content") + File.separator + Slug.makeSlug(identifier, true) + File.separator + Platform.getString(ARTIFACT_FOLDER, "artifact") else Platform.getString(CONTENT_FOLDER, "content") + File.separator + Slug.makeSlug(identifier, true) + File.separator + Platform.getString(ARTIFACT_FOLDER, "artifact")
```

Comment:
```suggestion
			val folder = if(StringUtils.isNotBlank(filePath.getOrElse(""))) filePath.get + File.separator + Platform.getString(CONTENT_FOLDER, "content") + File.separator + Slug.makeSlug(identifier, true) + File.separator + Platform.getString(ARTIFACT_FOLDER, "artifact") else Platform.getString(CONTENT_FOLDER, "content") + File.separator + Slug.makeSlug(identifier, true) + File.separator + Platform.getString(ARTIFACT_FOLDER, "artifact")
```
`should be change to:`

```suggestion
			val folder = if(StringUtils.isNotBlank(filePath.getOrElse(""))) {filePath.get + File.separator} + Platform.getString(CONTENT_FOLDER, "content") + File.separator + Slug.makeSlug(identifier, true) + File.separator + Platform.getString(ARTIFACT_FOLDER, "artifact") else Platform.getString(CONTENT_FOLDER, "content") + File.separator + Slug.makeSlug(identifier, true) + File.separator + Platform.getString(ARTIFACT_FOLDER, "artifact")
```